### ISSUE TITLE
lluminaBasecallsToSam READ_STRUCTURE cannot recognize an index including only molecular barcodes

### ISSUE THREAD

**Issue by misshie** [broadinstitute/picard#1320]
### Affected tool(s)
IlluminaBasecallsToSam READ_STRUCTURE

### Affected version(s)
- Latest public release version [2.19.0]

### Description 
`READ_STRUCTURE=100T8B10M100T` results the following error:

> "Exception in thread "main" picard.PicardException: Column label (BARCODE_2) unrecognized.  Library params(barcode params) can only contain the columns (OUTPUT, LIBRARY_NAME, SAMPLE_ALIAS, BARCODE, BARCODE_<X> where X is a positive integer) OR two letter RG tags!"

I wanted to describe "8bp index1 and 10bp index2".  According to  `picard/src/main/java/picard/illumina/parser/ReadStructure.java` (a switch in line 142-157), the number of indices are counted by `B`.  If `READ_STRUCTURE=100T8B0B10M100T` (using `0B10M` to describe index2 without sample-barcode) is accepted,  I can describe my read structure in READ_STRUCTURE.

**Comment by yfarjoun** [comment]
I'm confused since the READ_STRUCTURE you have in the description doesn't have 2 barcodes, and yet you claim that you want to describe a "8bp index1 and 10bp index2"....

**Comment by yfarjoun** [comment]
also, could you provide a stack-trace?

**Comment by yfarjoun** [comment]
why are you using `M` in your READ_STRUCTURE? that is for a UMI, or a molecular identifier, not for de-multiplexing. if that's what you have, you shouldn't be providing BARCODE_2, since the UMI is NOT a bar-code...it's "random" and so you do not want the data to be demultiplexed by it. 

in short, if you have two **sample barcodes** , you need two B elements and two columns in your params file.

if you have one **sample barcode** and one UMI you should have one B and one M element in your READ_STRUCTURE, and only one SAMPLE_BARCODE column in your PARAMS

**Comment by misshie** [comment]
I really appreciate your suggestion. 

>  in short, if you have two sample barcodes , you need two B elements and two columns in your params file.
> if you have one sample barcode and one UMI you should have one B and one M element in your READ_STRUCTURE, and only one SAMPLE_BARCODE column in your PARAMS

This worked well. I used "READ_STRUCTURE=100T8B10M100T"  and the "BARCODE" column containing barcodes such as "ACGTACGTNNNNNNNNNN" in the library_params file and ran ExtractIlluminaBarcods and IlluminaBasecallsToSam again. 

I made this mistake because we use Agilent's SureSelect XT Low Input and its manual instructed to run Hiseq2500 with 100, 8, 10, and 100 cycles for read1, index1 (sample barcodes) , index2 (random UMIs) and read2, respectively.
